src/fluent/watch.ts

```diff
@@ -382,53 +424,66 @@ export class Watcher<T extends GenericClass> {
 
         const { body } = response;
 
+        if (!body) {
+          throw new Error("No response body found");
```

Comment:
did we validate this is the correct behavior? what does controller runtime do here?

Reply:
Short answer is no, none of the failure scenarios has occurred in the wild bc it has not actually failed yet on a connection after running it through 2 [soak tests](https://github.com/defenseunicorns/pepr/actions/runs/11154315919/job/31003419390#step:16:733).  (We may want to try with UDS Core to put extra load on it)

I will get back to you with what controller runtime does, i looked it up a few months ago but dont remember.

Reply:
now remembering back on this, i think this is not needed, I was getting an [error](https://github.com/nodejs/undici/blob/87d7ccf6b51c61a4f4a056f7c2cac78347618486/docs/docs/api/Errors.md?plain=1#L16) which may been been my own fault. Will look at how controller runtime handles this